src/api/subscriptions.js

```diff
@@ -289,8 +289,38 @@ export class Subscriptions {
 
   /** @return {!Promise<ClientEventManagerApi>} */
   getEventManager() {}
+
+  /**
+   * Publishers participating in Showcase should call this with their own entitlements
+   * and entitlement related UI events.  SwG will automatically do this for Google
+   * sourced subscriptions and meters.
+   * @param {!Array<!PublisherEntitlement>} entitlements
+   */
+  setShowcaseEntitlements(entitlements) {}  
 }
 
+/** @enum {string} */
+export const PublisherEntitlementEvents = {
+  EVENT_SHOWCASE_UNLOCKED_BY_SUBSCRIPTION: 'EVENT_SHOWCASE_UNLOCKED_BY_SUBSCRIPTION', // Publisher managed subscriptions only
+  EVENT_SHOWCASE_UNLOCKED_BY_METER: 'EVENT_SHOWCASE_UNLOCKED_BY_METER',               // Publisher managed meters only
+  EVENT_SHOWCASE_UNLOCKED_FREE_PAGE: 'EVENT_SHOWCASE_UNLOCKED_FREE_PAGE',             // Must be reported by publisher
+  
+  EVENT_SHOWCASE_NO_ENTITLEMENTS_REGWALL: 'EVENT_SHOWCASE_NO_ENTITLEMENTS_REGWALL',  // Generated by swg-gaa or publisher
```

Comment:
This comment and the next one are confusing - just have the pub report them, worst case it's a duplicate which we can detect with the nonce,